File: springmvc/spring-springmvc-springdatajpa/src/main/java/com/jiangKlijna/web/app/ContextWrapper.java
package com.jiangKlijna.web.app;

import javax.servlet.ServletContext;

import org.springframework.context.ApplicationContext;
import org.springframework.web.context.ContextLoader;
import org.springframework.web.context.WebApplicationContext;
import org.springframework.web.context.support.WebApplicationContextUtils;

public class ContextWrapper {
	public WebApplicationContext getWebApplicationContext() {
		return ContextLoader.getCurrentWebApplicationContext();
	}

	public ServletContext getServletContext() {
		return getWebApplicationContext().getServletContext();
	}

	public ApplicationContext getApplicationContext() {
		return WebApplicationContextUtils.getWebApplicationContext(getServletContext());
	}
}

File: spring-springmvc-hibernate-kotlin/src/main/java/com/jiangKlijna/web/app/ContextWrapper.kt
package com.jiangKlijna.web.app

import javax.servlet.ServletContext

import org.springframework.context.ApplicationContext
import org.springframework.web.context.ContextLoader
import org.springframework.web.context.WebApplicationContext
import org.springframework.web.context.support.WebApplicationContextUtils

open class ContextWrapper {
    val webApplicationContext: WebApplicationContext
        get() = ContextLoader.getCurrentWebApplicationContext()

    val servletContext: ServletContext
        get() = webApplicationContext.servletContext

    val applicationContext: ApplicationContext
        get() = WebApplicationContextUtils.getWebApplicationContext(servletContext)
}
